cedar-lean/Cedar/Partial/Evaluator.lean

```diff
@@ -67,34 +83,61 @@ def hasAttr (v : Spec.Value) (a : Attr) (es : Partial.Entities) : Result Spec.Va
   let r ← Partial.attrsOf v (fun uid => .ok (es.attrsOrEmpty uid))
   .ok (r.contains a)
 
+/-- Analogous to Spec.hasAttr but for partial value and partial entities -/
+def hasAttrₚ (pv : Partial.Value) (a : Attr) (es : Partial.Entities) : Result Partial.Value := do
+  match pv with
+  | .value v₁ => do
+    let val ← Partial.hasAttr v₁ a es
+    .ok (.value val)
+  | .residual r => .ok (.residual (Partial.Expr.hasAttr r a)) -- TODO more precise: even though pv is a residual we may know concretely whether it contains the particular attr we care about
+
 /-- Analogous to Spec.getAttr but for partial entities -/
 def getAttr (v : Spec.Value) (a : Attr) (es : Partial.Entities) : Result Partial.Value := do
   let r ← Partial.attrsOf v es.attrs
   r.findOrErr a attrDoesNotExist
 
+/-- Analogous to Spec.getAttr but for partial value and partial entities -/
+def getAttrₚ (pv : Partial.Value) (a : Attr) (es : Partial.Entities) : Result Partial.Value := do
+    match pv with
+    | .value v₁ => Partial.getAttr v₁ a es
+    | .residual r => .ok (.residual (Partial.Expr.getAttr r a)) -- TODO more precise: pv will be a .residual if it contains any unknowns, but we might have a concrete value for the particular attr we care about
+
 /-- Analogous to Spec.bindAttr but for partial values -/
 def bindAttr (a : Attr) (res : Result Partial.Value) : Result (Attr × Partial.Value) := do
   let v ← res
   .ok (a, v)
 
+/-- Partial-evaluate a Var. No analogue in Spec.Evaluator; Spec.evaluate handles the `.var` case inline -/
+def evaluateVar (v : Var) (req : Partial.Request) : Result Partial.Value :=
```

Comment:
One more nit: I think it'd be nice and consistent if all the helper functions were called `evaluate*`.

Reply:
Renamed the functions with `p` subscripts to `evaluate*` (and also renamed `call` to `evaluateCall`) in the latest revision.